### PR TITLE
Topic/joint revolute unbounded

### PR DESCRIPTION
**Pull request by fvalenza**
. Added JointRevoluteUnbounded templated by the axis + unittest comparing 
. Fixed typo error in src/CMakelists
Related to #118

### CODE REVIEW

**Comment by jcarpent** [comment]
Related to #118.

File: src/multibody/joint/joint-revolute-unbounded.hpp [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+    }
+
+    ConfigVector_t random_impl() const
+    { 
+      ConfigVector_t result(ConfigVector_t::Random());
+      return result;
+    } 
+
+
+    ConfigVector_t randomConfiguration_impl(const ConfigVector_t & , const ConfigVector_t &  ) const throw (std::runtime_error)
```
Comment (jcarpent) [inline src/multibody/joint/joint-revolute-unbounded.hpp]: There is no throw in this function

File: src/multibody/joint/joint-revolute-unbounded.hpp [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+                  sin (theta0 + u * theta);
+      }
+      
+      return result; 
+    }
+
+    ConfigVector_t random_impl() const
+    { 
+      ConfigVector_t result(ConfigVector_t::Random());
+      return result;
```
Comment (jmirabel) [inline src/multibody/joint/joint-revolute-unbounded.hpp]: Shouldn't this be normalized ?

File: src/multibody/joint/joint-revolute-unbounded.hpp [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+      const double & c1 = qf(0), s1 = qf(1);
+
+      TangentVector_t result;
+      result << atan2 (s0*c1 - s1*c0, c0*c1 + s0*s1);
+      return result; 
+    } 
+
+    double distance_impl(const Eigen::VectorXd & q0,const Eigen::VectorXd & q1) const
+    { 
+      return difference_impl(q0,q1).norm();
```
Comment (jmirabel) [inline src/multibody/joint/joint-revolute-unbounded.hpp]: `fabs(difference_impl(q0,q1)[0])` is probably better.

**Comment by fvalenza** [comment]
- traits<JointRevoluteUnbounded>::Motion_t is now a MotionRevolute instead of a MotionRevoluteUnbounded to factorize the code 
- Same for Constraint_t
- Integrated joseph's remarks

**Comment by jmirabel** [comment]
A remark on this topic: the planar joint in HPP uses a unbounded rotation.

This joint better fits to application where the robot can move in a plane because we usually do not really care on counting the number of loops around z.

**Comment by jcarpent** [comment]
:+1:

**Pull request merged**